backtracking/rat_in_maze.py

```diff
@@ -4,101 +4,84 @@
-    We want to go from the start to the end. In this matrix zeroes represent walls
-    and ones paths we can use.
-    Parameters :
-        maze(2D matrix) : maze
+    In this problem, we have an n by n matrix, a start point, and an end point.
+    We want to go from the start to the end. In this matrix, ones represent walls,
+    and zeros represent paths we can use.
+
+    Parameters:
+        maze (2D matrix): The maze where 1 represents walls, and 0 represents paths.
+
     Returns:
-        Return: True if the maze has a solution or False if it does not.
+        bool: True if a solution exists, False otherwise.
+
     >>> maze = [[0, 1, 0, 1, 1],
     ...         [0, 0, 0, 0, 0],
     ...         [1, 0, 1, 0, 1],
     ...         [0, 0, 1, 0, 0],
     ...         [1, 0, 0, 1, 0]]
     >>> solve_maze(maze)
+    True
+    Path:
     [1, 0, 0, 0, 0]
     [1, 1, 1, 1, 0]
     [0, 0, 0, 1, 0]
     [0, 0, 0, 1, 1]
     [0, 0, 0, 0, 1]
-    True
 
     >>> maze = [[0, 1, 0, 1, 1],
     ...         [0, 0, 0, 0, 0],
     ...         [0, 0, 0, 0, 1],
     ...         [0, 0, 0, 0, 0],
     ...         [0, 0, 0, 0, 0]]
     >>> solve_maze(maze)
-    [1, 0, 0, 0, 0]
-    [1, 0, 0, 0, 0]
-    [1, 0, 0, 0, 0]
-    [1, 0, 0, 0, 0]
-    [1, 1, 1, 1, 1]
-    True
-
-    >>> maze = [[0, 0, 0],
-    ...         [0, 1, 0],
-    ...         [1, 0, 0]]
-    >>> solve_maze(maze)
-    [1, 1, 1]
-    [0, 0, 1]
-    [0, 0, 1]
-    True
-
-    >>> maze = [[0, 1, 0],
-    ...         [0, 1, 0],
-    ...         [1, 0, 0]]
-    >>> solve_maze(maze)
-    No solution exists!
     False
-
-    >>> maze = [[0, 1],
-    ...         [1, 0]]
-    >>> solve_maze(maze)
     No solution exists!
-    False
     """
     size = len(maze)
-    # We need to create solution object to save path.
+    # We need to create a solution object to save the path.
     solutions = [[0 for _ in range(size)] for _ in range(size)]
     solved = run_maze(maze, 0, 0, solutions)
+
     if solved:
-        print("\n".join(str(row) for row in solutions))
+        print("Path:")
+        for row in solutions:
+            print(row)
     else:
         print("No solution exists!")
     return solved
 
 
 def run_maze(maze: list[list[int]], i: int, j: int, solutions: list[list[int]]) -> bool:
     """
-    This method is recursive starting from (i, j) and going in one of four directions:
+    This method is recursive, starting from (i, j) and going in one of four directions:
     up, down, left, right.
-    If a path is found to destination it returns True otherwise it returns False.
+    If a path is found to the destination, it returns True; otherwise, it returns False.
+
     Parameters:
-        maze(2D matrix) : maze
-        i, j : coordinates of matrix
-        solutions(2D matrix) : solutions
+        maze (2D matrix): The maze where 1 represents walls, and 0 represents paths.
+        i, j (int): Coordinates in the matrix.
```

Comment:
Please do not repeat simple types (e.g. int, bool, etc.) that are already in the function signature because the types in the function signature are tested by `mypy` and readers will get confused if the function signature gets modified and the comments do not or vice versa.